src/main/java/com/whitesource/remediation/encoder/Encode.java

```diff
@@ -0,0 +1,209 @@
+package com.whitesource.remediation.encoder;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.SystemUtils;
+import org.owasp.esapi.codecs.UnixCodec;
+import org.owasp.esapi.codecs.WindowsCodec;
+
+/**
+ * Remediation Solver static class written by WhiteSource with the community ❤. Here you can find
+ * wrapper functions to secure unsafe operations in your code.
+ */
+public class Encode {
+
+  /**
+   * Encoding operating system parameters.
+   *
+   * @param params The parameters for the operating systems.
+   * @return Encoded parameters.
+   */
+  public static String OSParameterEncoder(@NonNull final String params)
+          throws UnsupportedOperationException {
+    if (SystemUtils.IS_OS_WINDOWS) {
+      return Utils.esapiEncoder(new WindowsCodec(), params);
+    } else if (SystemUtils.IS_OS_UNIX) {
+      return Utils.esapiEncoder(new UnixCodec(), params);
+    }
+
+    throw new UnsupportedOperationException("Unsupported encoder for operating system");
+  }
+
+  /**
+   * Encoding content for logs.
+   *
+   * @param contents arrays {@link Object} contains all the contents.
+   * @return encoded log content.
+   */
+  public static String[] multiLogContentEncoder(@NonNull final Object[] contents) {
```

Comment:
In order to simplify analysis in core, those functions should be overloading i.e should have the same name "log content encoder".